src/nvim/window.c

```diff
@@ -5812,7 +5812,8 @@ void win_setheight_win(int height, win_T *win)
     msg_col = 0;
 
     if (*p_spk != 'c') {
-      win_fix_scroll(true);
+      // cursor only needs to be fixed if it's in the current window
+      win_fix_scroll(win == curwin);
```

Comment:
This will conflict with #23884. I'm also not so sure it's correct anymore.
It had been a while since I worked on 'splitkeep', and the way the `resize` argument is used is confusing. But even if `win != curwin` here, `curwin`'s height might have changed if it is in the same frame as `win`, in which case the cursor _would_ need to be adjusted.

Reply:
It would be good to grasp the problem you're trying to fix with an actual reproducible `test.lua` but perhaps you could work around the behavior by deferring the jump to definition instead?

Reply:
In my particular test, the current window's height didn't change. I jumped to a new buffer loaded in the same window to a certiain location (`vim.lsp.buf.jump_to_location`). After that an autocmd ran that adjusted the height of an unrelated window, but the fix cursor code then ***fixed*** the cursor position of the current window/buffer, which is not correct. Result being that I'm not at the location I jumped to.

What would be a better fix for this? I'm not that fmailiar with the splitkeep code.

Reply:
It seemed to adjust the cursor of the current window to the height of the window I was changing the height for

Reply:
Wouldn't that have happened _before_ your change to `nvim_win_set_height()`, but not anymore?

Reply:
Before my change, the current window would indeed have been set to the window that's changing height, so I guess in that case it's indeed needed to fix the cursor?

But with my new change, the current window isn't always the window we're changing.

Reply:
I'll try yo create a minimal config to trigger the issue